tests/scripts/restore-mon-quorum-using-osds.sh

```diff
@@ -0,0 +1,319 @@
+    done
+    restore_manifests
+
+    kubectl -n "${NAMESPACE}" scale deployment rook-ceph-operator --replicas=1
+    kubectl -n "${NAMESPACE}" delete pod -l app=rook-ceph-operator
+    wait_pods rook-ceph-operator 1 Running
+
+    DESIRED_MGR_COUNT=$(kubectl get deployment -n "${NAMESPACE}" -l app=rook-ceph-mgr --no-headers 2>/dev/null | wc -l)
+    kubectl -n "${NAMESPACE}" delete pod -l app=rook-ceph-mgr
+    wait_pods rook-ceph-mgr $DESIRED_MGR_COUNT Running
+}
+
+function init() {
+    if ! command -v kubectl; then
+        echo "'kubectl' must exist" >&2
+        exit 1
+    fi
+    if ! command -v jq;  then
+        echo "'jq' must exist" >&2
+        exit 1
+    fi
+    if ! kubectl get pod >/dev/null 2>&1; then
+        echo "couldn't connect to your kubernetes cluster" >&2
+        exit 1
+    fi
+
+    TOOL_COUNT=$(kubectl get pod -n "${NAMESPACE}" -l app=rook-ceph-tools --no-headers 2>/dev/null | wc -l)
+    if [ ${TOOL_COUNT} != 1 ]; then
+      echo "toolbox pod must exist" >&2
+      exit 1
+    fi
+    TOOL_DEPLOYMENT=$(kubectl get deployment -n "${NAMESPACE}" -l app=rook-ceph-tools -o json | jq -r '.items[0].metadata.name')
+
+    if [ "${RESTORE}" = 0 ]; then
+        mkdir -p "${BACKUP_DIR}"
+        if [ $(ls "${BACKUP_DIR}" | wc -l) != 0 ]; then
+            echo "backup directory must be empty: ${BACKUP_DIR}" >&2
+            exit 1
+        fi
+    else
+        if [ ! -d "${BACKUP_DIR}" -o $(ls "${BACKUP_DIR}" | wc -l) = 0 ]; then
+            echo "backup directory must exist and must contain backup data" >&2
+            exit 1
+        fi
+    fi
+
+    TMP_MON_STORE="${TMP_DIR}"/mon-store
+    rm -rf "${TMP_MON_STORE}"
+    mkdir -p "${TMP_MON_STORE}"
+
+    MONS=$(kubectl get deployment -n "${NAMESPACE}" -l app=rook-ceph-mon -o json | jq -r '.items[].metadata.name')
+    DESIRED_MON_COUNT=$(echo ${MONS} | wc -w)
+    if [ ${DESIRED_MON_COUNT} = 0 ]; then
+        echo "the desired number of MON should be >= 1: ${DESIRED_OSD_COUNT}" >&2
+        exit 1
+    fi
+    OSDS=$(kubectl get deployment -n "${NAMESPACE}" -l app=rook-ceph-osd -o json | jq -r '.items[].metadata.name')
+    DESIRED_OSD_COUNT=$(echo ${OSDS} | wc -w)
+    if [ $DESIRED_OSD_COUNT = 0 ]; then
+        echo "the desired number of OSD should be >= 1: ${DESIRED_OSD_COUNT}" >&2
+        exit 1
+    fi
+
+    # MON_IDS should be sorted by IP addresses due to the limitation of ceph-monstore-tool. See the following Ceph's issue for more detail.
+    # https://tracker.ceph.com/issues/49158
+    MON_IDS=$(kubectl get svc -n "${NAMESPACE}" -l app=rook-ceph-mon -o json | jq -r '.items[] | [.metadata.name, .spec.clusterIP] | @sh' | tr -d \' |  sort -t . -k 2,2n -k 3,3n -k 4,4n | cut -d' ' -f1 |  cut -d'-' -f4)
+    OSD_IDS=$(kubectl get -n "${NAMESPACE}" po -l app=rook-ceph-osd -o json | jq -r '.items[].metadata.labels["osd"]')
```

Comment:
There is no container with the `osd` label for me, it must change to `ceph-osd-id`.